### PR TITLE
fixes process_DEProblem

### LINKED ISSUE #1484: DAEProblem fails
Maybe I'm completely wrong but can it be that the following code is not correct and  `u0defs` and `pdefs`  should be switched?
Is the ordering of the `merge` correct? Since `defs = defaults(sys)`, the user input e.g. `u0map` will be always overruled. However, testing this it seems to work for e.g. `ODEProblem`.
https://github.com/SciML/ModelingToolkit.jl/blob/029addab54a82361ef4a3d97e732f4779cb2334f/src/systems/diffeqs/abstractodesystem.jl#L562-L579

Since `pdefs` is used for the parameters:
https://github.com/SciML/ModelingToolkit.jl/blob/029addab54a82361ef4a3d97e732f4779cb2334f/src/systems/diffeqs/abstractodesystem.jl#L589

This one errors (see example bellow). Should `defaults` be `du0defs` (which is not defined)?
https://github.com/SciML/ModelingToolkit.jl/blob/029addab54a82361ef4a3d97e732f4779cb2334f/src/systems/diffeqs/abstractodesystem.jl#L584

MWE:
```julia
using ModelingToolkit, Sundials
@parameters t
D = Differential(t)
@variables x[1:2](t) = zeros(2) 
@variables y(t) = 0.0
eqs= [
    D(x[1]) ~ x[2]
    D(x[2]) ~ -x[1] - 0.5 * x[2] + 1.0
    y ~ 0.9 * x[1] + x[2]
]
@named model = ODESystem(eqs, t, vcat(x, y), [])
prob = DAEProblem(sys, Pair[], Pair[], (0, 50))
sol = solve(prob, IDA())
```

```julia
ERROR: MethodError: no method matching iterate(::typeof(ModelingToolkit.defaults))
Closest candidates are:
  iterate(::Union{LinRange, StepRangeLen}) at \julia\base\range.jl:826
  iterate(::Union{LinRange, StepRangeLen}, ::Integer) at C\julia\base\range.jl:826
  iterate(::T) where T<:Union{Base.KeySet{<:Any, <:Dict}, Base.ValueIterator{<:Dict}} at \julia\base\dict.jl:695
  ...
Stacktrace:
 [1] isempty(itr::Function)
   @ Base .\essentials.jl:775
 [2] varmap_to_vars(varmap::Vector{Pair}, varlist::Vector{Term{Real, Nothing}}; defaults::Function, check::Bool, toterm::Function)
   @ ModelingToolkit \.julia\packages\ModelingToolkit\Abs2j\src\variables.jl:40
 [3] process_DEProblem(constructor::Type, sys::ODESystem, u0map::Vector{Pair}, parammap::SciMLBase.NullParameters; implicit_dae::Bool, du0map::Vector{Pair}, version::Nothing, tgrad::Bool, jac::Bool, checkbounds::Bool, sparse::Bool, simplify::Bool, linenumbers::Bool, parallel::Symbolics.SerialForm, eval_expression::Bool, kwargs::Base.Pairs{Symbol, Bool, Tuple{Symbol}, NamedTuple{(:has_difference,), Tuple{Bool}}})
   @ ModelingToolkit \.julia\packages\ModelingToolkit\Abs2j\src\systems\diffeqs\abstractodesystem.jl:584
```

```
[961ee093] ModelingToolkit v8.5.2
[c3572dad] Sundials v4.9.3
```

Also this cries for a method:
```julia
function mergedefaults(defaults, varmap, vars)
    if varmap isa Dict
        merge(varmap, defaults)
    elseif eltype(varmap) <: Pair
        merge(Dict(varmap), defaults)
    elseif eltype(varmap) <: Number
        merge(Dict(zip(vars, varmap)), defaults)
    else
        defaults
    end
end

eqs = equations(sys)
iv = get_iv(sys)
dvs = states(sys)
ddvs = map(Differential(iv), dvs)
ps = parameters(sys)
defs = defaults(sys)
pdefs = mergedefaults(defs, parammap, ps)
u0defs = mergedefaults(defs, u0map, dvs)
du0defs = mergedefaults(defs, du0map, ddvs)
```

### PR DESCRIPTION
**Pull request by ValentinKaisermayer**
closes #1484

### CODE REVIEW

File: src/systems/diffeqs/abstractodesystem.jl [hunk 1]
```diff
@@ -1,870 +1,868 @@
-function calculate_tgrad(sys::AbstractODESystem;
-                         simplify=false)
```
Comment (ChrisRackauckas) [inline src/systems/diffeqs/abstractodesystem.jl:2]: How come the diff is so messed up?

**Comment by ChrisRackauckas** [comment]
Did you save the file in a different encoding or something? This cannot be reviewed with a diff that just says the whole file changed 😅

File: src/systems/diffeqs/abstractodesystem.jl (hunk 1 above)
Comment (ValentinKaisermayer) [reply src/systems/diffeqs/abstractodesystem.jl:2]: I'm not sure. In my VSCode diff it works. Any suggestions how to resolve this?

Comment (ValentinKaisermayer) [reply src/systems/diffeqs/abstractodesystem.jl:2]: I ticked off whitespaces in the diff viewer

Comment (ChrisRackauckas) [reply src/systems/diffeqs/abstractodesystem.jl:2]: Can you submit a version that doesn't change the whitespaces? Any formatting change should be a separate PR.

**Comment by ValentinKaisermayer** [comment]
Sry, it seems VSCode did removed all trailing whitespaces.

**Pull request closed without merging**